website/source/docs/guides/monitoring-telegraf.html.md

```diff
@@ -0,0 +1,299 @@
+
+# Configuring Telegraf
+
+Besides acting as a statsd agent, Telegraf can collect additional metrics of its own. Telegraf itself ships
+with a wide range of [input plugins][telegraf-input-plugins] to collect data from lots of sources.
+
+We're going to enable some of the most common ones to monitor CPU, memory, disk I/O, networking, and process status, as these are useful for debugging Consul cluster issues.
+
+The `telegraf.conf` file starts with global options:
+
+```ini
+[agent]
+  interval = "10s"
+  flush_interval = "10s"
+  omit_hostname = false
+```
+
+We set the default collection interval to 10 seconds and ask Telegraf to include a `host` tag in each metric.
+
+As mentioned above, Telegraf also allows you to set additional tags on the metrics that pass through it. In this
+case, we are adding tags for the server role and datacenter. We can then use these tags in Grafana to filter
+queries (for example, to create a dashboard showing only servers with the `consul-server` role, or only servers
+in the `us-east-1` datacenter).
+
+```ini
+[global_tags]
+  role = "consul-server"
+  datacenter = "us-east-1"
+```
+
+Next, we set up a statsd listener on UDP port 8125, with instructions to calculate percentile metrics and to
+parse DogStatsD-compatible tags, when they're sent:
+
+```ini
+[[inputs.statsd]]
+  protocol = "udp"
+  service_address = ":8125"
+  delete_gauges = true
+  delete_counters = true
+  delete_sets = true
+  delete_timings = true
+  percentiles = [90]
+  metric_separator = "_"
+  parse_data_dog_tags = true
+  allowed_pending_messages = 10000
+  percentile_limit = 1000
+```
+
+The full reference to all the available statsd-related options in Telegraf is [here][telegraf-statsd-input].
+
+Now we can configure inputs for things like CPU, memory, network I/O, and disk I/O. Most of them don't require any configuration, but make sure the `interfaces` list in `inputs.net` matches the interface names you see in `ifconfig`.
+
+```ini
+[[inputs.cpu]]
+  percpu = true
+  totalcpu = true
+  collect_cpu_time = false
+
+[[inputs.disk]]
+  # mount_points = ["/"]
+  # ignore_fs = ["tmpfs", "devtmpfs"]
+
+[[inputs.diskio]]
+  # devices = ["sda", "sdb"]
+  # skip_serial_number = false
+
+[[inputs.kernel]]
+  # no configuration
+
+[[inputs.linux_sysctl_fs]]
+  # no configuration
+
+[[inputs.mem]]
+  # no configuration
+
+[[inputs.net]]
+  interfaces = ["enp0s*"]
+
+[[inputs.netstat]]
+  # no configuration
+
+[[inputs.processes]]
+  # no configuration
+
+[[inputs.swap]]
+  # no configuration
+
+[[inputs.system]]
+  # no configuration
+```
+
+Another useful plugin is the [procstat][telegraf-procstat-input] plugin, which reports metrics for processes you select:
+
+```ini
+[[inputs.procstat]]
+  pattern = "(consul)"
+```
+
+Finally, Telegraf even includes a [plugin][telegraf-consul-input] to monitor Consul agents.
```

Comment:
I don't think this plugin generally monitors the agent, as much as it monitors the health checks associated with the agent. It's kind of surprising so you may want to clarify what it does. https://github.com/influxdata/telegraf/tree/master/plugins/inputs/consul

Reply:
Clarified.